src/api.c

```diff
@@ -117,7 +117,12 @@ yaml_string_join(
 YAML_DECLARE(int)
 yaml_stack_extend(void **start, void **top, void **end)
 {
-    void *new_start = yaml_realloc(*start, ((char *)*end - (char *)*start)*2);
+    void *new_start;
+
+    if ((char *)*end - (char *)*start >= INT_MAX / 2)
```

Comment:
Is there a reason not to calculate

```c
(char *)*end - (char *)*start
```

Once and use it here and below? I know it shouldn't change very much, but I feel like it'll be cleaner, especially if we give it a very clear name.